### PR TITLE
[wxwidgets] Fix #4756

### LINKED ISSUE #4756: wxwidgets installed but not detected by FindwxWidgets.cmake
wxWidgets installed with simple `vcpkg install wxwidgets:x64-windows-static` without problems.
Simple minimal CMake project
```cmake
cmake_minimum_required(VERSION 3.12)
project(test)
find_package(wxWidgets REQUIRED
    COMPONENTS core qa html xml aui adv stc webview base scintilla)
```
[generates error](https://ci.appveyor.com/project/slodki/appveyor-tests#L39):
```
CMake Error at C:/Program Files (x86)/CMake/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find wxWidgets (missing: wxWidgets_LIBRARIES
  wxWidgets_INCLUDE_DIRS)
Call Stack (most recent call first):
  C:/Program Files (x86)/CMake/share/cmake-3.12/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  C:/Program Files (x86)/CMake/share/cmake-3.12/Modules/FindwxWidgets.cmake:963 (find_package_handle_standard_args)
  C:/Tools/vcpkg/scripts/buildsystems/vcpkg.cmake:247 (_find_package)
  CMakeLists.txt:6 (find_package)
```
because libraries are installed in non-standard directory (${WX_ROOT_DIR}/lib in place of ${WX_ROOT_DIR}/lib/vc141...). See
https://github.com/Kitware/CMake/blob/be9ad8279a29893943b342cc5ffdbf9868fb1b4c/Modules/FindwxWidgets.cmake#L544

### PR DESCRIPTION
**Pull request by ras0219**
Alternative approach compared to #13023

- What does your PR fix?
Fixes #4756

I'm able to successfully compile https://docs.wxwidgets.org/trunk/overview_helloworld.html with the following CMakeLists.txt:
```cmake
cmake_minimum_required(VERSION 3.17)
project(foo)

add_executable(foo foo.cpp)

find_package(wxWidgets REQUIRED COMPONENTS core qa html xml aui adv stc webview base scintilla)
include(${wxWidgets_USE_FILE})
target_link_libraries(foo ${wxWidgets_LIBRARIES})
```
The source also required modification to add `wxIMPLEMENT_WXWIN_MAIN_CONSOLE` since our built copy of wxwidgets targets GUIs.

### CODE REVIEW

**Comment by marekr** [comment]
This seems like a very awkward hack. Such as removing WXUSINGDLL in all header files making them rather...unique compared to all normal uses of wxwidgets.

You are supposed to define WXUSINGDLL in your application.

**Comment by ras0219** [comment]
@marekr I'm really happy you've been able to work around these issues with a custom file!

However, in upstream vcpkg,

1) We really don't want to force every user to write a custom `FindwxWidgets.cmake`
2) We have to support more buildsystems than just CMake

Because vcpkg always deploys exactly one version of wxwidgets (either DLL or static) we can statically encode the linkage in the package, which removes a source of error on the consumption side. If the user correctly set the macro in their project, there will be no observable difference in the build. This is a technique we apply broadly across vcpkg for libraries to handle ABI-affecting macros.

**Review by NancyLi1013 — COMMENTED** [review]
The `control `file for `wxchartdir` has been delated. `vcpkg.json` seems missing in this PR.

**Comment by NancyLi1013** [comment]
The failure on osx:

```
There should be no bin\ directory in a static build, but /Users/vagrant/Data/packages/wxwidgets_x64-osx/bin is present.
There should be no debug\bin\ directory in a static build, but /Users/vagrant/Data/packages/wxwidgets_x64-osx/debug/bin is present.
If the creation of bin\ and/or debug\bin\ cannot be disabled, use this in the portfile to remove them

    if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
    endif()
```

**Review by BillyONeal — APPROVED** [review]

File: ports/wxwidgets/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "wxwidgets",
+  "version-string": "3.1.4",
+  "port-version": 5,
```
Comment (NancyLi1013) [inline ports/wxwidgets/vcpkg.json:4]:
```suggestion
  "port-version": 4,
```

File: ports/wxwidgets/portfile.cmake [hunk 2]
```diff
@@ -61,16 +57,34 @@ if(DLLS)
...
+    endif()
+    # Remove install prefix from setup.h to ensure package is relocatable
+    string(REGEX REPLACE "\n#define wxINSTALL_PREFIX [^\n]*" "\n#define wxINSTALL_PREFIX \"\"" _contents "${_contents}")
+    file(WRITE "${INC}" "${_contents}")
+endforeach()
 
-file(INSTALL ${SOURCE_PATH}/docs/licence.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
\ No newline at end of file
+file(COPY ${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake DESTINATION ${CURRENT_PACKAGES_DIR}/share/wxwidgets)
+configure_file(${CMAKE_CURRENT_LIST_DIR}/usage ${CURRENT_PACKAGES_DIR}/share/wxwidgets/usage COPYONLY)
```
Comment (NancyLi1013) [inline ports/wxwidgets/portfile.cmake:89]:
```suggestion
file(COPY ${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
configure_file(${CMAKE_CURRENT_LIST_DIR}/usage ${CURRENT_PACKAGES_DIR}/share/${PORT}/usage COPYONLY)
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
image

**Pull request merged**